### PR TITLE
display only for simulation-in-hardware

### PR DESCRIPTION
**Pull request by romain-chiap**
This pull request allows the states of the quadrotor from [simulation-in-hardware](https://dev.px4.io/master/en/simulation/simulation-in-hardware.html) to be displayed in jMAVSim.

By starting with the flag `-displayonly`, jMAVSim enters a mode where the simulation engine is disabled. I created a class `MAVLinkDisplayOnly` that extends `MAVLinkHILSystem`. It sends nothing to the autopilot and passes states from autopilot to simulator. The readed states from `HIL_STATE_QUATERNION` are the position and attitude. 

The documentation is also updated, by starting with the flag `-h`.

### CODE REVIEW

File: src/me/drton/jmavsim/Simulator.java [hunk 1]
```diff
@@ -196,7 +197,7 @@ public Simulator() throws IOException, InterruptedException {
             connCommon.addSkipMessage(schema.getMessageDefinition("HIL_ACTUATOR_CONTROLS").id);
             connCommon.addSkipMessage(schema.getMessageDefinition("HIL_SENSOR").id);
             connCommon.addSkipMessage(schema.getMessageDefinition("HIL_GPS").id);
-            connCommon.addSkipMessage(schema.getMessageDefinition("HIL_STATE_QUATERNION").id);
+            // connCommon.addSkipMessage(schema.getMessageDefinition("HIL_STATE_QUATERNION").id);
```
Comment (bkueng) [inline src/me/drton/jmavsim/Simulator.java]: Do you need this?

File: src/me/drton/jmavsim/MAVLinkDisplayOnly.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ * MAVLinkDisplayOnly is MAVLink bridge between AbstractVehicle and autopilot connected via MAVLink.
+ * MAVLinkDisplayOnly should have the same sysID as the autopilot, but different componentId.
+ * It reads HIL_STATE_QUATERNION from the MAVLink and displays the vehicle position and attitude.
+ * @author Romain Chiappinelli
+ */
+public class MAVLinkDisplayOnly extends MAVLinkHILSystem {
+
+    private boolean fistMsg=true;       // to detect the first MAVLink message
```
Comment (bkueng) [inline src/me/drton/jmavsim/MAVLinkDisplayOnly.java]:
```suggestion
    private boolean firstMsg=true;       // to detect the first MAVLink message
```

File: src/me/drton/jmavsim/MAVLinkDisplayOnly.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * MAVLinkDisplayOnly is MAVLink bridge between AbstractVehicle and autopilot connected via MAVLink.
+ * MAVLinkDisplayOnly should have the same sysID as the autopilot, but different componentId.
+ * It reads HIL_STATE_QUATERNION from the MAVLink and displays the vehicle position and attitude.
+ * @author Romain Chiappinelli
+ */
+public class MAVLinkDisplayOnly extends MAVLinkHILSystem {
```
Comment (bkueng) [inline src/me/drton/jmavsim/MAVLinkDisplayOnly.java]: Since this overrides/ignores most functionality of MAVLinkHILSystem, I'd rather have this inherit from `MAVLinkSystem` instead.
And as `Simulator` uses part of the `MAVLinkHILSystem` interface, we can introduce a `MAVLinkHILSystemBase` between `MAVLinkSystem` and `MAVLinkHILSystem`, which declares the interface required for `Simulator`.
What do you think?

Comment (romain-chiap) [reply src/me/drton/jmavsim/MAVLinkDisplayOnly.java]: That makes sense, I will think about it.

Comment (romain-chiap) [reply src/me/drton/jmavsim/MAVLinkDisplayOnly.java]: I am checking the feasibility, it becomes difficult as `Visualizer3D` needs `MAVLinkHILSystem`, same for `hilSystem.gotHilActuatorControls()`. 
1. We can have one variable of type `MAVLinkHILSystemBase`, instantiate `MAVLinkDisplayOnly` or `MAVLinkHILSystem` and cast them when needed:
`MAVLinkHILSystemBase hilSystem`
`((MAVLinkHILSystem)hilSystem).gotHilActuatorControls()`
2. We can have two distinct objects, one `MAVLinkHILSystem` and one `MAVLinkDisplayOnly`

I think the second option is preferable. What do you think ?

Comment (bkueng) [reply src/me/drton/jmavsim/MAVLinkDisplayOnly.java]: The third option is to add a `gotHilActuatorControls()` to `MAVLinkHILSystemBase`, and override it as needed. I see that as the cleanest way, or am I missing something?

**Comment by romain-chiap** [comment]
@bkueng 
I would like to do a followup on this pull request. Does that look good to you? 

I will update the Devguide soon.

File: src/me/drton/jmavsim/MAVLinkHILSystemBase.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package me.drton.jmavsim;
+
+import me.drton.jmavlib.mavlink.MAVLinkSchema;
+import me.drton.jmavsim.vehicle.AbstractVehicle;
+
+/**
+ * MAVLinkHILSystem is MAVLink bridge between AbstractVehicle and autopilot connected via MAVLink.
```
Comment (bkueng) [inline src/me/drton/jmavsim/MAVLinkHILSystemBase.java]: This needs to be updated.

File: src/me/drton/jmavsim/Simulator.java [hunk 5]
```diff
@@ -325,7 +332,7 @@ public void run() {
 
                     System.out.println("Shutting down...");
                     if (hilSystem != null) {
-                        hilSystem.endSim();
+                        (hilSystem).endSim();
```
Comment (bkueng) [inline src/me/drton/jmavsim/Simulator.java]: Why did you add brackets here (same below)?
